### ISSUE TITLE
Question: Custom Errors

### ISSUE THREAD

**Issue by gbabula** [philbooth/check-types.js#71]
__Question: Custom Errors__

`check.assert(value, message)` throws an `Error` if value is `false`, setting message on the `Error` instance. Is there a simple way to extend `assert` to throw a custom Error type while maintaining all the core functionality?

I have a few workarounds in mind, just wanted to check if y'all thought about this use case first.

---

_BTW I love the library, great work._ 👍

**Comment by philbooth** [comment]
Not right now, but I'm open to adding it.

Would an extra, optional, `ErrorType` argument to each of the assertion methods work for you or were you thinking of something different?

**Comment by gbabula** [comment]
@philbooth that's exactly what I was thinking, seems like a low risk addition that should be useful.

```js
Check.assert.instanceStrict(
    someArgument,
    SomeArgumentClass,
    errorMessage,
    CustomErrorType
);
```

**Comment by philbooth** [comment]
Released as version `7.1.0`, let me know if you get any problems with it!

**Comment by gbabula** [comment]
@philbooth thank you so much!

Did you publish to NPM? I wanted to write a few unit tests to verify the functionality but it looks like the latest release there is `7.0.1`.

Edit: guess you were in the middle of doing it :-) I have `7.1.0` now, going to verify everything and report back in this thread.
